### ISSUE TITLE
Black flashing on my projector

### ISSUE THREAD

**Issue by eblanca** [mpv-player/mpv#6500]
### mpv version and platform
I'm using mpv from master branch and ffmpeg lib from their 4.0x branch.
Built the whole thing with the mpv-build script

### Reproduction steps
I recently created a custom /etc/X11/xorg.conf file for my laptop (i5+integrated_gpu and discrete radeon - launched "sudo Xorg -configure") taking care in activating tearfree option. Now, with almost any video content, my projector wrongly flashes black frames.

### Expected behavior
Smooth playing.

### Actual behavior
The projector (hdmi connected) flashes a black screen among normal frames several times a minute. This is very annoying and ONLY HAPPENS in full screen and ONLY with particular encodings (weird, I know).

### Log file
https://0x0.st/zz74.json (h264 stream played with my xorg.conf config)
https://0x0.st/zz7J.json (h265 stream played with my xorg.conf config)
https://0x0.st/zz7y.json (dvb (h262) stream player with my xorg.conf config - THIS DOESN'T FLASH AT ALL)
https://0x0.st/zz7t.json (h264 stream played without xorg.conf THIS DOESN'T FLASH EITHER)

Again, flashing only happens in full screen mode.

https://0x0.st/zz7w.conf (my xorg.conf)

**Comment by lachs0r** [comment]
Most likely not an mpv issue. Make sure you’re not using the Intel Xorg DDX 
(which is buggy all the time on all hardware) but instead use modesetting.
You might not need TearFree with that one either but YMMV.

There’s nothing indicating any problems in the mpv logs you’ve posted from 
what I can tell. Xorg logs would probably help, but this isn’t really the 
place to get help with that.

**Comment by eblanca** [comment]
> https://0x0.st/zz7y.json (dvb (h262) stream player with my xorg.conf config - THIS DOESN'T FLASH AT ALL)

That's a typo: the stream is h264 FullHD.

**Comment by eblanca** [comment]
Anyway, this idea the root cause is a driver only, cannot explain why the black flashing does not affect some videos.

> Make sure you’re not using the Intel Xorg DDX (which is buggy all the time on all hardware) but instead use modesetting.

How can I check this?
Will it suffice removing /etc/X11/xorg.conf ?

**Comment by lachs0r** [comment]
Undefined behavior and bugs are weird like that, especially on the graphics driver side. It’s unlikely mpv really has anything to do with it.

The xorg.conf you’ve posted shows that it’s using the “intel” driver, which is the DDX I mentioned. Try changing that to “modesetting”.

**Comment by zc62** [comment]
https://wiki.archlinux.org/index.php/Intel_graphics

How is your xorg.conf generated? My experience is that, if without xorg.conf or any conf files in xorg.conf.d, xorg starts normally, then do not use any conf files. Nvidia needs a configuration file generated by `nvidia-xconfig`, while Intel most likely does not need any.

**Comment by eblanca** [comment]
> How is your xorg.conf generated?

`sudo Xorg -configure`

> My experience is that, if without xorg.conf or any conf files in xorg.conf.d, xorg starts normally,

I can confirm this, I used no xorg.conf until a couple of weeks ago, and I'm going back to remove the existing one very soon, since it brings no noticeable improvement.

The issue is, I always try to fine tune my system because with this laptop (3 years old, with debian, ubuntu and lmde I tried) I always had stuttering with videos, while never ever had with previous (slower) PCs. But this is another story.
Well, customizing xorg.conf is no way, I learnt.
Thank you anyway.
